src/refactor_nrepl/externs.clj

```diff
@@ -0,0 +1,28 @@
+(ns refactor-nrepl.externs
+  (:import java.net.URL
+           org.projectodd.shimdandy.ClojureRuntimeShim
+           java.net.URLClassLoader
+           java.util.ArrayList))
+
+(def rt (atom nil))
+
+(defn create-runtime []
+  (let [urls [(URL. "file:libs/shimdandy-impl-1.1.0.jar")
+              (URL. "file:libs/clojure-1.6.0.jar")
+              (URL. "file:libs/pomegranate-0.3.1-SNAPSHOT.jar")]
+        cl (URLClassLoader. (into-array URL urls))]
+    (reset! rt (ClojureRuntimeShim/newRuntime cl "refactor-nrepl"))))
+
+(defn add-dependencies [dep repos]
+  (when-not @rt
+    (create-runtime))
+  (.require @rt (into-array String ["cemerick.pomegranate"]))
```

Comment:
Ah, good point - it does have to see the lib/ dir for shimdandy to create a runtime. I'm not sure why it can't load pomegranate then. Can you share a gist of the stack you get?

Normal delegation has the classloader ask it's parent *first*, then search itself if the parent doesn't have it (see http://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html). So the parent's version of clojure will always be used. This also means that any namespaces required in the shimdandy runtime will be added to the internal lib cache in the "global" clojure runtime, but the classes would not be visible to that runtime when shimdandy's classloader isn't active. I'm not sure what sort of problems that would cause.